website/static/js/markdown.js

```diff
@@ -30,12 +31,18 @@ var bootstrapTable = function(md) {
 var oldMarkdownList = function(md) {
     md.block.ruler.after('hr', 'pyMarkdownList', pymarkdownList);
 };
+var mfrURL = window.contextVars.mfrURL;
+var osfURL = window.contextVars.osfURL;
 
 // Full markdown renderer for views / wiki pages / pauses between typing
 var markdown = new MarkdownIt('commonmark', {
     highlight: highlighter,
     linkify: true
-})
+    }).use(require('markdown-it-mfr'), {
+        type: 'osf',
+        pattern: /^http(?:s?):\/\/(?:www\.)?[a-zA-Z0-9 .:]{1,}\/render\?url=http(?:s?):\/\/[a-zA-Z0-9 .:]{1,}\/([a-zA-Z0-9]{5})\/\?action=download|(^[a-zA-Z0-9]{5}$)/,
+        formatUrl: function(assetID) { return mfrURL + 'render?url='+ osfURL + assetID + '/?action=download%26mode=render'; },
```

Comment:
Why is &mode=render needed here?

Reply:
Not exactly sure, but when it's removed MFR errors with the following: ` aiohttp.errors.ClientOSError: [Errno None] Cannot connect to host localhost:7777 ssl:False [Can not connect to localhost:7777 [None]]`